pkg/apis/management.cattle.io/v3/machine_types.go

```diff
@@ -400,9 +400,10 @@ type CloudCredential struct {
 }
 
 type CloudCredentialSpec struct {
-	DisplayName        string              `json:"displayName"`
-	Description        string              `json:"description,omitempty"`
-	S3CredentialConfig *S3CredentialConfig `json:"s3credentialConfig,omitempty"`
+	DisplayName             string                   `json:"displayName"`
+	Description             string                   `json:"description,omitempty"`
+	S3CredentialConfig      *S3CredentialConfig      `json:"s3credentialConfig,omitempty"`
+	AlibabaCredentialConfig *AlibabaCredentialConfig `json:"alibabacredentialConfig,omitempty"`
```

Comment:
I don't believe this is the right way of doing this. Are you doing this because the lack of a node driver means we aren't generating the dynamic schema for this?

Reply:
I don't think this is a viable solution long term. We are planning on introducing a v1 api for cloud credentials. I would prefer an approach where we implement something similar to the `createCredSchema`. @kinarashah WDYT?

Reply:
added the schema generation in management data addition logic